lib/queue.js

```diff
@@ -650,9 +652,19 @@ Queue.prototype.getMetadata = function getMetadata(queue) {
     }
     // Extract meta-data
     var metadata = {};
+    var rawHeaderCounter = 0;
     for(var field in res.headers) {
+      rawHeaderCounter++;
       if (/x-ms-meta-/.test(field)) {
-        metadata[field.substr(10)] = res.headers[field];
+        // Metadata names must adhere to the naming rules for C# identifiers, which are case-insensitive,
+        // meaning that you can set,for example, a metadata with the following form:
+        // {
+        //    applicationName: 'fast-azure-blob-storage'
+        // }
+        // In order to return the original metadata name, the header names should be read from response.rawHeaders.
+        // That is because 'https' library returns the response headers with lowercase.
+        var key = res.rawHeaders[rawHeaderCounter * 2 - 2];
```

Comment:
Sorry, I can't remember or find the structure of `response.rawHeaders`.  The comment is great, but could you include a brief explanation of why you're using the `rawHeaderCount * 2 - 2` value?